plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateCommandWrapper.java

```diff
@@ -132,8 +132,9 @@ Use VIR_DOMAIN_XML_SECURE (value = 1) prior to v1.0.0.
             vmsnapshots = libvirtComputingResource.cleanVMSnapshotMetadata(dm);
 
             Map<String, MigrateCommand.MigrateDiskInfo> mapMigrateStorage = command.getMigrateStorage();
+            final boolean migrateStorage = MapUtils.isNotEmpty(mapMigrateStorage);
```

Comment:
This variable does not need to be final. It does not bring any benefits here. 

You needed to assign the result of the first `MapUtils.isNotEmpty(mapMigrateStorage)` because the method `replaceStorage` might change the status of `mapMigrateStorage`, right?